Add service discovery for IONOS Cloud (#10514)
prometheus/prometheus: docs/configuration/configuration.md

http_sd_configs:
  [ - <http_sd_config> ... ]

# List of IONOS service discovery configurations.

> ionos_sd_configs:

Reviewer: This should be twice on the page, once for alerting and once for scraping.

Author: Thanks and fixed.

  [ <tls_config> ]
```

### `<ionos_sd_config>`

IONOS SD configurations allows retrieving scrape targets from
[IONOS Cloud](https://cloud.ionos.com/) API. This service discovery uses the
first NICs IP address by default, but that can be changed with relabeling. The
following meta labels are available on all targets during
[relabeling](#relabel_config):

* `__meta_ionos_server_availability_zone`: the availability zone of the server
* `__meta_ionos_server_cpu_family`: the CPU family of the server
* `__meta_ionos_server_datacenter_id`: the datacenter ID the server is deployed
  to
* `__meta_ionos_server_id`: the ID of the server
* `__meta_ionos_server_ip`: comma seperated list of IPs assigned to the server
* `__meta_ionos_server_lan_id`: comma seperated list of LANs' IDs attached to the
  server
* `__meta_ionos_server_lifecycle`: the lifecycle state of the server resource
* `__meta_ionos_server_name`: the name of the server
* `__meta_ionos_server_nic_id`: comma seperated list of the NIC IDs attached to
  the server
* `__meta_ionos_server_primary_ip`: the primary IP address of the server

> * `__meta_ionos_server_primary_lan_id`: the primary LAN's ID of the server

Reviewer: ```suggestion
* `__meta_ionos_server_primary_lan_id`: the ID of the primary LAN
```

> * `__meta_ionos_server_primary_nic_id`: the primary NIC's ID of the server

Reviewer: ```suggestion
* `__meta_ionos_server_primary_nic_id`: the ID of the primary NIC
```

* `__meta_ionos_server_state`: the execution state of the server
* `__meta_ionos_server_type`: the type of the server

> The labels below depend on the targets boot medium and are mutually exclusive:

Reviewer: ```suggestion
```

I would just mix them with the above labels

Author: Thanks for the suggestions and updated the documentation.

> * `__meta_ionos_server_boot_cdrom_id`: the CD-ROM's ID the server is booted

Reviewer: ```suggestion
* `__meta_ionos_server_boot_cdrom_id`: the ID of the CD-ROM the server is booted from
```

  from

> * `__meta_ionos_server_boot_image_id`: the ID of the image or snapshot the

Reviewer: ```suggestion
* `__meta_ionos_server_boot_image_id`: the ID of the boot image or snapshot
```

  server is booted from

> * `__meta_ionos_server_boot_volume_id`: the volume's ID the server is booted

Reviewer: ```suggestion
* `__meta_ionos_server_boot_volume_id`: the ID of the boot volume
```

  from

```yaml

> # The IONOS datacenter ID in which entities should be discovered.

Reviewer: ```suggestion
# The unique ID of the data center.
```

Can we replace datacenter by data center as that is what is used in IONOS documentation.
